torch/_subclasses/fake_tensor.py

```diff
@@ -1306,6 +1309,107 @@ def tolist(self):
     __torch_function__ = torch._C._disabled_torch_function_impl
 
 
+@dataclass(frozen=True)
```

Comment:
This class is essentially duplicated with another that I added in the previous code cache work. I'll combine them, just lemme know where they belong conceptually.Is there an appropriate "utils" module, for example? Or maybe it's fine if they go here.